app/src/main/java/ch/epfl/sdp/blindly/recording/AudioLibraryAdapter.kt

```diff
@@ -0,0 +1,218 @@
+    private fun toggleLayout(isExpanded: Boolean, v: View, layoutExpand: RelativeLayout): Boolean {
+        if (isExpanded) {
+            RecordAnimations.expand(layoutExpand)
+        } else {
+            RecordAnimations.collapse(layoutExpand)
+        }
+        return isExpanded
+    }
+
+    // Create new views (invoked by the layout manager)
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
+        // Create a new view, which defines the UI of the list item
+        val view = LayoutInflater.from(context).inflate(R.layout.audio_recording, null)
+        return ViewHolder(view)
+    }
+
+    // Replace the contents of a view (invoked by the layout manager)
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
+        // Get element from your dataset at this position and replace the
+        // contents of the view with that element
+        viewHolder.recordName.text = recordList[position].name
+        viewHolder.recordDuration.text = recordList[position].durationText
+        viewHolder.nameDurationLayout.setOnClickListener {
+            val show: Boolean = toggleLayout(!recordList[position].isExpanded, it, viewHolder.expandableLayout)
+            recordList[position].isExpanded = show
+        }
+
+        val playBar = viewHolder.playBar
+        val playTimer = viewHolder.playTimer
+        val remainingTimer = viewHolder.remainingTimer
+        val playPauseButton = viewHolder.playPauseButton
+        val movePlayBarThread = createPlayBarThread(playBar)
+
+        setCountDownTimer(remainingTimer)
+
+        viewHolder.playPauseButton.setOnClickListener {
+            if (isPlayerStopped) {
+                createMediaPlayer(recordList[position].filePath)
+                remainingTimer.base = SystemClock.elapsedRealtime() + mediaPlayer!!.duration.toLong()
+
+                mediaPlayer?.setOnCompletionListener {
+                    mediaPlayer?.stop()
+
+                    setCompletedAudioView(playTimer, remainingTimer, playPauseButton)
+                }
+            }
+            if (!mediaPlayer!!.isPlaying) {
+                mediaPlayer?.start()
+
+                if (!isPlayerPaused) {
+                    playTimer.base = SystemClock.elapsedRealtime()
+                }
+                setPlayView(playTimer, remainingTimer, playBar, movePlayBarThread, playPauseButton)
+            } else {
+                mediaPlayer?.pause()
+
+                setPauseView(playTimer, remainingTimer, playPauseButton)
+            }
+        }
+
+        viewHolder.selectButton.setOnClickListener {
+            mediaPlayer?.release()
+            saveRecording(position)
+            startProfileFinished()
+        }
+    }
+
+    private fun saveRecording(position: Int) {
+        val filePath = recordList[position].filePath
+        val newName = "PresentationAudio.3gp"
```

Comment:
Why not use @string resource here ?

Reply:
I don't think it is a good idea for a file name.